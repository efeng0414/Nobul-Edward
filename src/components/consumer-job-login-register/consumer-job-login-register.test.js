import React from "react";
import renderer from "react-test-renderer";
import ConsumerJobLoginRegister from "./index";
import toJSON from "enzyme-to-json";

describe("Consumer Job Login/Register modal Component", () => {
  const componentProps = {
    intl: global.getIntl(),
    visible: true,
    onCancel: () => {},
    onLogin: () => {},
    onRegister: () => {}
  };
  const componentJSX = <ConsumerJobLoginRegister {...componentProps} />;

  describe("Component Snapshot", () => {
    const componentSnapshot = toJSON(global.Enzyme.shallow(componentJSX));
    it("should match existing snapshot", () => {
      expect(componentSnapshot).toMatchSnapshot();
    });
  });

  describe("Register with Email click event", () => {
    it("should call onRegister function when the button is clicked", () => {
      const componentMount = global.Enzyme.shallow(componentJSX);
      const mockFn = jest.fn();
      componentMount.setProps({ onRegister: mockFn });
      componentMount.find(".loginRegister__register-button").simulate("click");
      expect(mockFn).toHaveBeenCalled();
    });
  });

  describe("Login click event", () => {
    it("should call onLogin function when the button is clicked", () => {
      const componentMount = global.Enzyme.shallow(componentJSX);
      const mockFn = jest.fn();
      componentMount.setProps({ onLogin: mockFn });
      componentMount.find(".loginRegister__login-button").simulate("click");
      expect(mockFn).toHaveBeenCalled();
    });
  });
});
