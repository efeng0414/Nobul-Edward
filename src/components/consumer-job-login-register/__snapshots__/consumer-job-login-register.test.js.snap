// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consumer Job Login/Register modal Component Component Snapshot should match existing snapshot 1`] = `
<Modal
  cancelButtonDisabled={false}
  confirmLoading={false}
  footer={null}
  maskTransitionName="fade"
  okButtonDisabled={false}
  okType="primary"
  onCancel={[Function]}
  prefixCls="ant-modal"
  transitionName="zoom"
  visible={true}
  width={520}
>
  <div
    className="loginRegister"
  >
    <Button
      block={false}
      className="loginRegister__register-button"
      ghost={false}
      loading={false}
      onClick={[Function]}
      prefixCls="ant-btn"
      size="large"
      type="primary"
    >
      Sign up with Email
    </Button>
    <div
      className="loginRegister__line"
    />
    <div
      className="loginRegister__login"
    >
      <span>
        Already have an account? 
      </span>
      <button
        className="loginRegister__login-button"
        onClick={[Function]}
      >
        Log in
      </button>
    </div>
  </div>
</Modal>
`;
